### ISSUE TITLE
pals: experience gets worse as you acquire more pals

### ISSUE THREAD

**Issue by Fang-** [Fang-/suite#6]
This is the opposite of good!

- [x] Page generation becomes slow because we're rendering many sigils. Probably best to start serving these as their own files, at `/pals/~/palfun-foslup.svg` or wherever, so we can tell the browser to just cache them. Initial overhead might be worse due to separate requests, but should be smoother sailing afterwards.
  - Maybe we want to start doing this only if we have more than a certain number of pals? Probably not worth it though.
  - Wonder if we can separate this logic out nicely without upending the entire `webui` setup again. Perhaps `/pals/sigil.svg?p=palfun-foslup&c=decafe`? Would let use give `immutable` caching headers too.
- [x] List becomes hard to navigate and see changes to. I'm not keen on storing date-ordered peer lists, but certainly we can support filtering by state and/or tag?

**Comment by vcavallo** [comment]
> certainly we can support filtering by state and/or tag?

What’s wrong with ctrl-f ? 
:P

**Comment by Fang-** [comment]
Performance concerns have been alleviated by f9fbfc0 and ffffa74. Together they reduce page rendering times in "many pals with contact details" cases from tens of seconds to tens of _milliseconds_.

(The update containing those changes is blocked on memory pressure concerns, currently figuring out how safe it is to release this.)

While ctrl-f is fine, some minimal filtering options would still be nice to have, so this issue remains open.

**Comment by Fang-** [comment]
As of f4fe27d (in pals v1.0.11) we have some basic filtering affordances, resolving the second point on this issue.
